lib/bookingsync/engine/credentials_resolver.rb

```diff
@@ -0,0 +1,18 @@
+class BookingSync::Engine::CredentialsResolver
+  attr_accessor :host
+
+  def initialize(host)
+    @host = host
+  end
+
+  def call
+    if application = ::Application.find_by_host(host)
```

Comment:
how about passing application as an argument? That way, we would decouple this object from database

Reply:
This just move the DB coupling from this resolver to the initializer of the omniauth provider, not sure I see the benefit.